pilot/pkg/config/kube/ingressv1/controller.go

```diff
@@ -0,0 +1,340 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ingress provides a read-only view of Kubernetes ingress resources
+// as an ingress rule configuration type store
+package ingress
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"sort"
+	"time"
+
+	knetworking "k8s.io/api/networking/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	ingressinformer "k8s.io/client-go/informers/networking/v1"
+	listerv1 "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/pilot/pkg/model"
+	kubecontroller "istio.io/istio/pilot/pkg/serviceregistry/kube/controller"
+	"istio.io/istio/pkg/config"
+	"istio.io/istio/pkg/config/constants"
+	"istio.io/istio/pkg/config/mesh"
+	"istio.io/istio/pkg/config/schema/collection"
+	"istio.io/istio/pkg/config/schema/collections"
+	"istio.io/istio/pkg/config/schema/gvk"
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/queue"
+	"istio.io/pkg/env"
+)
+
+// In 1.0, the Gateway is defined in the namespace where the actual controller runs, and needs to be managed by
+// user.
+// The gateway is named by appending "-istio-autogenerated-k8s-ingress" to the name of the ingress.
+//
+// Currently the gateway namespace is hardcoded to istio-system (model.IstioIngressNamespace)
+//
+// VirtualServices are also auto-generated in the model.IstioIngressNamespace.
+//
+// The sync of Ingress objects to IP is done by status.go
+// the 'ingress service' name is used to get the IP of the Service
+// If ingress service is empty, it falls back to NodeExternalIP list, selected using the labels.
+// This is using 'namespace' of pilot - but seems to be broken (never worked), since it uses Pilot's pod labels
+// instead of the ingress labels.
+
+// Follows mesh.IngressControllerMode setting to enable - OFF|STRICT|DEFAULT.
+// STRICT requires "kubernetes.io/ingress.class" == mesh.IngressClass
+// DEFAULT allows Ingress without explicit class.
+
+// In 1.1:
+// - K8S_INGRESS_NS - namespace of the Gateway that will act as ingress.
+// - labels of the gateway set to "app=ingressgateway" for node_port, service set to 'ingressgateway' (matching default install)
+//   If we need more flexibility - we can add it (but likely we'll deprecate ingress support first)
+// -
+
+var schemas = collection.SchemasFor(
+	collections.IstioNetworkingV1Alpha3Virtualservices,
+	collections.IstioNetworkingV1Alpha3Gateways)
+
+// Control needs RBAC permissions to write to Pods.
+
+type controller struct {
```

Comment:
Is this all new code or was it pulled out of git history?

Reply:
Essentially `cp -r ingress ingressv1; sed s/v1beta1/v1/g`. No major changes